### DISCUSSION TITLE
Buffalo theme/ example project

### DISCUSSION THREAD

**Discussion by piti2** [gobuffalo/buffalo#2251] [Q&A]
Hi there,

are there buffalo theme or example projects one can start with for easy learning and see working examples of a live example site/project. thanks

**Comment by sio4** [comment]
You can find some sample applications built upon Buffalo here (somewhat old by the way):
- https://github.com/gobuffalo/toodo
- https://github.com/orgs/gobuffalo/repositories?q=example

However, Buffalo does not have any theme since Buffalo is a web application development framework and the theming is out of scope basically.

**Reply by piti2** [reply]
Thank you @sio4 for demonstrating how to convert a static site to buffalo in the prior example. However, because hugo is a golang program and bufallow is the same, why would I need to create a static site and then manually change the routing and all the links and partials before converting it to bufallow? Wouldn't it be ideal to keep hugo for html, CSS, and JS generation but adding buffalow partials to hugo partials to make routing and other things easier, rather than doing it manually for each and every link?

-------
i am saying hugo as an example but it can be any other golang static site generator

**Reply by sio4** [reply]
The thing both Hugo and Buffalo are written in golang is not relevant. Also, you don't need to create a static site and then change them into Buffalo. They are totally different things with different purposes. My answers (here and for the issue you filed #2250) were based on your question that how to "migrate from hugo to buffalo".

What is your goal when considering Buffalo? Is it a web application that interacts with users dynamically? Or is it a static website? If you are looking for something for a static website, Buffalo is not a good choice.

I think it could be easy to discuss if you clearly explain your situation and the goal.

**Reply by piti2** [reply]
Hi @sio4 

I'm only interested in a dynamic site that allows users to sign up/login and leave comments on blog posts. Hugo's static site generator was used to create the blog post. Now that I understand the dynamic aspect of Buffel, I'm stumped as to how I can fast create pages when dynamic functionality accounts for just 5% of the website and the rest is static. Is it necessary for me to write everything from scratch, or can I use another program to generate static material quickly? 

So, for example, a blog is entirely static, with comments as the only dynamic feature. If I create everything in Bufallo from the ground up, I'll have to figure out how to produce responsive images as well as multilingual functionality etc.

So what you recommend i should do i mention Hugo as i have already build the fully optimised frontend and need dynamic functionality for token validation and routing user after login etc.

thanks for taking the time to read and help

**Reply by sio4** [reply]
I feel like your goal became clearer to me.

> I'm only interested in a dynamic site that allows users to sign up/login and leave comments on blog posts. Hugo's static site generator was used to create the blog post. Now that I understand the dynamic aspect of Buffel, I'm stumped as to how I can fast create pages when dynamic functionality accounts for just 5% of the website and the rest is static. Is it necessary for me to write everything from scratch, or can I use another program to generate static material quickly?

Actually, everything is depending on what you want in detail. For me, I don't want to spend time creating my own commenting system and running a web application for my own blog, so my suggestion is to use a comment service. Personally, I am using disqus.com, and also Facebook could be a good alternative.

If you really want to have your own "users" just for leaving a comment on your blog, you can build a commenting system from the scratch and you may be able to connect them to your static blog running with any other web proxies such as Nginx on your own server, or any static site service provider including Github Pages or Netlify. Generally, if your blog is still public and no need to login to the site just for reading, you don't need to combine them together into a single service built with Buffalo.

By the way, now I feel like this topic is not directly related to Buffalo. :-)

**Reply by piti2** [reply]
> I feel like your goal became clearer to me.
> 
> > I'm only interested in a dynamic site that allows users to sign up/login and leave comments on blog posts. Hugo's static site generator was used to create the blog post. Now that I understand the dynamic aspect of Buffel, I'm stumped as to how I can fast create pages when dynamic functionality accounts for just 5% of the website and the rest is static. Is it necessary for me to write everything from scratch, or can I use another program to generate static material quickly?
> 
> Actually, everything is depending on what you want in detail. For me, I don't want to spend time creating my own commenting system and running a web application for my own blog, so my suggestion is to use a comment service. Personally, I am using disqus.com, and also Facebook could be a good alternative.
> 
> If you really want to have your own "users" just for leaving a comment on your blog, you can build a commenting system from the scratch and you may be able to connect them to your static blog running with any other web proxies such as Nginx on your own server, or any static site service provider including Github Pages or Netlify. Generally, if your blog is still public and no need to login to the site just for reading, you don't need to combine them together into a single service built with Buffalo.
> 
> By the way, now I feel like this topic is not directly related to Buffalo. :-)

Hi @sio4 

the comments are unrelevant it can be something else also, My question is how can i build frontend of the dynamic webapp in buffalo. for example click here this is an image. is there builder i can use or i have to hand code each button image language etc? and if i handcode can buffalo optimise code and images to make the webapp faster?

thanks

**Reply by sio4** [reply]
Everything is only depending on what you want to make and the way you want to use it. Buffalo is a web application development framework, the rest totally depends on the developers. For more details on how it works, how to start, and so on, please check the documentation: https://gobuffalo.io/documentation/
